src/js/abstract/hocs/with-data-seleniumid.js

```diff
@@ -0,0 +1,29 @@
+const withDataSeleniumId = Base =>
+  /**
+   * Base class {WithBaseGlobal}. This class extends the {Base} and
+   * applies styles globally by injecting them within `<head>` section.
+   * This is the recommended approach.
+   */
+  class WithDataSeleniumId extends Base {
+
+    connectedCallback() {
+      if (super.connectedCallback) {
+        super.connectedCallback();
+      }
+
+      const { constructor: { tagName } } = this;
```

Comment:
Great.
Now we can discuss if we really still need additional checks if this feature shall be enabled.
IMHO not necessarily.

More important is that we document how Selenium selects `<custom-elements>` and how it interacts with its children.

@LucaMele 
I see one challenge here, ideally it's possible to enable it from outside of any component, for example by:
- `window.__enableDataSeleniumId` ???
- one more HTML attribute for each component, like `selenium-id-enabled`

Reply:
@AndyOGo i do like the idea of having `window.__enableDataSeleniumId` setting.

Reply:
Alright.

So @michel-ho let's go with that